diff --git a/arch/arm64/boot/dts/amlogic/gxbb_p200.dts b/arch/arm64/boot/dts/amlogic/gxbb_p200.dts
index 803045e..646b469 100644
--- a/arch/arm64/boot/dts/amlogic/gxbb_p200.dts
+++ b/arch/arm64/boot/dts/amlogic/gxbb_p200.dts
@@ -12,6 +12,7 @@
 #include <dt-bindings/reset/aml_gxbb.h>
 #include "mesongxbb.dtsi"
 / {
+	amlogic-dt-id = "gxb_p200_1g";
 	model = "Amlogic";
 	compatible = "amlogic, Gxbb";
 	interrupt-parent = <&gic>;
diff --git a/arch/arm64/boot/dts/amlogic/gxbb_p200_2G.dts b/arch/arm64/boot/dts/amlogic/gxbb_p200_2G.dts
index 1c4c324..60f3902 100644
--- a/arch/arm64/boot/dts/amlogic/gxbb_p200_2G.dts
+++ b/arch/arm64/boot/dts/amlogic/gxbb_p200_2G.dts
@@ -12,6 +12,7 @@
 #include <dt-bindings/reset/aml_gxbb.h>
 #include "mesongxbb.dtsi"
 / {
+	amlogic-dt-id = "gxb_p200_2g";
 	model = "Amlogic";
 	compatible = "amlogic, Gxbb";
 	interrupt-parent = <&gic>;
