diff --git a/language/English/strings.po b/language/English/strings.po
index 27e4a4e..5b032b2 100644
--- a/language/English/strings.po
+++ b/language/English/strings.po
@@ -837,3 +837,15 @@ msgstr ""
 msgctxt "#32391"
 msgid "Enable LCD Driver Support"
 msgstr ""
+
+msgctxt "#60003"
+msgid "Clear Memory Cache"
+msgstr ""
+
+msgctxt "#60004"
+msgid "Press to clean"
+msgstr ""
+
+msgctxt "#60005"
+msgid "Press to clean"
+msgstr ""
diff --git a/src/resources/lib/modules/system.py b/src/resources/lib/modules/system.py
index 2273898..aa2a4cb 100644
--- a/src/resources/lib/modules/system.py
+++ b/src/resources/lib/modules/system.py
@@ -207,6 +207,20 @@ class system:
                             },
                         },
                     },
+                'cacheclear': {
+                    'order': 9,
+                    'name': 60003,
+                    'settings': {
+                        'cache_clear': {
+                            'name': 60004,
+                            'value': '0',
+                            'action': 'cache_clear',
+                            'type': 'button',
+                            'InfoText': 60005,
+                            'order': 1,
+                            },
+                        },
+                    },
                 }
 
             self.keyboard_layouts = False
@@ -636,6 +650,19 @@ class system:
             self.oe.set_busy(0)
             self.oe.dbg_log('system::reset_oe', 'ERROR: (' + repr(e) + ')', 4)
 
+    def cache_clear(self, listItem=None):
+        try:
+            self.oe.dbg_log('system::cache_clear', 'enter_function', 0)
+            self.oe.set_busy(1)
+            self.oe.execute('/usr/sbin/cacheclear')
+            #self.oe.winOeMain.close()
+            time.sleep(1)
+            self.oe.set_busy(0)
+            self.oe.dbg_log('system::cache_clear', 'exit_function', 0)
+        except Exception, e:
+            self.oe.set_busy(0)
+            self.oe.dbg_log('system::cache_clear', 'ERROR: (' + repr(e) + ')', 4)
+
     def ask_sure_reset(self, part):
         try:
             self.oe.dbg_log('system::ask_sure_reset', 'enter_function', 0)
